core/light-client/src/lib.rs

```diff
@@ -381,7 +374,22 @@ where
 
 		Ok(())
 	}
+}
+
+pub trait LightClientState<Block: BlockT> {
```

Comment:
With the extraction of this trait, we can remove at some places the traitbound:

`NumberFor<ParentchainBlock>: BlockNumberOps`